cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -461,13 +462,29 @@ class descriptor : public detail::descriptor_base<Task> {
 
 /// @tparam Task   Tag-type that specifies the type of the problem to solve. Can
 ///                be :expr:`task::v1::classification` or :expr:`task::v1::regression`.
+namespace dt = dal::decision_tree;
+template <typename Task>
+struct decision_tree_task_map;
+
+template <>
+struct decision_tree_task_map<task::classification> {
+    using tree_task_t = dt::task::classification;
+};
+
+template <>
+struct decision_tree_task_map<task::regression> {
+    using tree_task_t = dt::task::regression;
+};
+
 template <typename Task = task::by_default>
 class model : public base {
     static_assert(detail::is_valid_task_v<Task>);
     friend dal::detail::pimpl_accessor;
 
 public:
     using task_t = Task;
+    using dtree_task_t = typename decision_tree_task_map<task_t>::tree_task_t;
+    using visitor_t = std::shared_ptr<typename dt::detail::node_visitor_iface<dtree_task_t>>;
```

Comment:
Is there reason to expose it publicly?

Reply:
I'm strongly against exposing `visitor_t` as it contradicts with the template parameters accepted by `traverse_dfs`/`traverse_bfs`.

Reply:
moved to private